gsd/test/test_fl.py

```diff
@@ -1211,7 +1315,34 @@ def test_pending_index_entries(tmp_path):
         f.flush()
 
         f.end_frame()
+        f.flush()
 
         # All test chunks should be present in the file.
         for i in range(16):
             assert f.chunk_exists(name=str(i), frame=1)
+
+
+@pytest.mark.parametrize('flush', [False, True])
+def test_expand(tmp_path, open_mode, flush):
+    """Test index expansion."""
+    with gsd.fl.open(
+        name=tmp_path / 'test_expand.gsd',
+        mode=open_mode.write,
+        application='test_expand',
+        schema='none',
+        schema_version=[1, 2],
+    ) as f:
+        N_ENTRIES = 1024
+
+        for i in range(N_ENTRIES):
+            data = numpy.array([i], dtype=numpy.int64)
+            f.write_chunk(name='data', data=data)
+            f.end_frame()
+            if flush and (i & 0xF == 0):
+                f.flush()
+
+        f.flush()
+        for i in range(N_ENTRIES):
+            expected_data = numpy.array([i], dtype=numpy.int64)
```

Comment:
I'm not sure if this is valid/the correct thing to test, but it could be nice to validate the expansion for different sized inputs.

```suggestion
            data = numpy.array([i] * i, dtype=numpy.int64)
            f.write_chunk(name='data', data=data)
            f.end_frame()
            if flush and (i & 0xF == 0):
                f.flush()

        f.flush()
        for i in range(N_ENTRIES):
            expected_data = numpy.array([i] * i, dtype=numpy.int64)
```

Reply:
Sure.